tools/python/runfiles/runfiles.py

```diff
@@ -106,6 +138,17 @@ def Create(env=None):
   return None
 
 
+def _FindPythonRunfilesRoot():
```

Comment:
I'm very wary of re-implementing logic to find the runfiles. The logic to find it is very non-trivial in the stub template:

https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt

If you look at runfiles.py here, you see that the Create() function doesn't try to guess any of that -- it just expects someone to have set one of the RUNFILES_* env vars.

Actually, er -- this newly added logic looks completely independent of that? The self._python_runfiles_root is never actually used anywhere? Am I missing something?

Reply:
I added a comment explaining why I am using this logic: Essentially, we just want to know what was in `sys.path[0]` when the Python stub ran. If it's unlikely that a `py_binary` would modify `sys.path[0]` at runtime, I could just use that here. What do you think?

Reply:
Runfiles libraries are given either the runfiles directory or the path of the runfiles manifest file and have to work with that. While the runfiles directory can be reverse engineered from the path of the runfiles manifest, that logic is non-trivial (hence the stub template) and may not work with `--build_python_zip`.

I found it both easier and more reliable to reverse engineer the Python runfiles directory from an actual Python file instead - after all, we know that the stub template set everything up for us as otherwise we wouldn't be running.